src/main/java/com/adyen/httpclient/HttpURLConnectionClient.java

```diff
@@ -43,11 +43,14 @@
      * config is used to obtain basic auth username, password and User-Agent
      */
     @Override
-    public String request(String requestUrl, String requestBody, Config config) throws IOException, HTTPClientException {
+    public String request(String requestUrl, String requestBody, Config config, boolean isApiKeySupported) throws IOException, HTTPClientException {
         HttpURLConnection httpConnection = createRequest(requestUrl, config.getApplicationName());
+        if(isApiKeySupported)
+        {
+            setApiKey(httpConnection, config.getApiKey());
+        }
         setBasicAuthentication(httpConnection, config.getUsername(), config.getPassword());
```

Comment:
if api key is required, we shouldn't set basic auth headers

Reply:
Basic authentication was set because core system checks username and password if API key is not present. Today discussed with core team and they confirmed that we should ignore username and password if API key is required. Hence incorporating the suggested changes and not setting basic auth if API key is required. thanks.